src/safeposix/dispatcher.rs

```diff
@@ -152,6 +167,159 @@ pub extern "C" fn dispatcher(cageid: u64, callnum: i32, arg1: Arg, arg2: Arg, ar
         GETPID_SYSCALL => {
             check_and_dispatch!(cage.getpid_syscall,)
         }
+        SOCKET_SYSCALL => {
+            check_and_dispatch!(cage.socket_syscall, interface::get_int(arg1), interface::get_int(arg2), interface::get_int(arg3))
+        }
+
+        BIND_SYSCALL => {
+            let addrlen = get_onearg!(interface::get_uint(arg3));
+            let addr = get_onearg!(interface::get_sockaddr(arg2, addrlen));
+            check_and_dispatch!(cage.bind_syscall, interface::get_int(arg1), Ok::<&interface::GenSockaddr, i32>(&addr))
+        }
+
```

Comment:
Minor comment:
There seem to be some irregular empty lines between syscalls. It might be better to always have them or never have them.